packages/zql/src/query/query-impl.ts

```diff
@@ -98,32 +97,25 @@ export function staticParam(
 
 export const SUBQ_PREFIX = 'zsubq_';
 
+export const defaultFormat = {singular: false, relationships: {}} as const;
+
 export abstract class AbstractQuery<
   TSchema extends Schema,
   TTable extends keyof TSchema['tables'] & string,
   TReturn = PullRow<TTable, TSchema>,
-> implements AdvancedQuery<TSchema, TTable, TReturn>
+> implements Query<TSchema, TTable, TReturn>
 {
   readonly #schema: TSchema;
   readonly #tableName: TTable;
   readonly #ast: AST;
-  readonly #format: Format;
+  readonly format: Format;
   #hash: string = '';
 
-  constructor(
-    schema: TSchema,
-    tableName: TTable,
-    ast: AST = {table: tableName},
-    format?: Format | undefined,
-  ) {
+  constructor(schema: TSchema, tableName: TTable, ast: AST, format: Format) {
     this.#schema = schema;
     this.#tableName = tableName;
     this.#ast = ast;
-    this.#format = format ?? {singular: false, relationships: {}};
-  }
-
-  get format(): Format {
-    return this.#format;
+    this.format = format;
```

Comment:
I think caused a regression.  Nate is hitting errors because of an undefined format.  

![image](https://github.com/user-attachments/assets/7b03a0bc-f7cd-4a3a-87d3-ed112ebf1d4b)

One way format can end up undefined here are these calls https://github.com/rocicorp/mono/blob/bf275086f4c906b6cb7e801a5eaa740e9570f3e8/packages/zql/src/query/query-impl.ts#L420
https://github.com/rocicorp/mono/blob/bf275086f4c906b6cb7e801a5eaa740e9570f3e8/packages/zql/src/query/query-impl.ts#L456

unsure if there are others.

Not sure why TypeScript isn't catching this.

Reply:
Thanks for the fix